jekyll/_cci2/server-3-operator-extending-internal-volumes.adoc

```diff
@@ -0,0 +1,172 @@
+
+For postgres
+[source,bash]
+----
+circleci-user ~ $ kubectl get pvc data-postgresql-0
+NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+data-postgresql-0   Bound    pvc-c2a2d97b-2b7d-47d3-ac77-d07c76c995a3   8Gi        RWO            gp2            1d
+----
+
+For mongodb
+[source,bash]
+----
+circleci-user ~ $ kubectl get pvc datadir-mongodb-0
+NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+datadir-mongodb-0   Bound    pvc-58a2274c-31c0-487a-b329-0062426b5535   8Gi        RWO            gp2            1d
+----
+
+We can also confirm this capacity is made available to a database by checking the size of it's data directory.
+
+For postgres the directory is `/bitnami/postgresql`. We can confirm it's size using the command below.
+
+[source,bash]
+----
+circleci-user ~ $ kubectl exec postgresql-0 -- df -h /bitnami/postgresql
+Filesystem      Size  Used Avail Use% Mounted on
+/dev/nvme4n1    7.8G  404M  7.4G   3% /bitnami/postgresql
+----
+
+For mongodb the directory is `/bitnami/mongodb`.
+[source,bash]
+----
+circleci-user ~ $ kubectl exec mongodb-0 -- df -h /bitnami/mongodb
+Filesystem      Size  Used Avail Use% Mounted on
+/dev/nvme1n1    7.8G  441M  7.4G   3% /bitnami/mongodb
+----
+
+From the examples above, the capacities are still 8Gi. Lets increase the size to 10Gi
+
+=== Step 1 - Confirm Volume Expansion is Allowed
+First let's confirm that volume expansion is allowed in our cluster.
+
+[source,bash]
+----
+circleci-user ~ $ kubectl get sc
+NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   false                  1d
+----
+
+As we can see, our default storage class does not allow volume expansion. However, changing this is quite easy with the `kubectl patch` command.
+
+[source,bash]
+----
+circleci-user ~ $ kubectl patch sc gp2 -p '{"allowVolumeExpansion": true}'
+storageclass.storage.k8s.io/gp2 patched
+circleci-user ~ $ kubectl get sc
+NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   true                  1d
+----
+
+Now we may proceed to expand our volumes.
+
+=== Step 2 - Delete the Database's Stateful Set
+In this step we wish to delete the stateful set which controls our database pod, however we do not want to delete the pod itself as this would cause downtime. The command below will delete the referenced database's stateful set without deleting the pod. Not to worry, in the steps below we will be redeploying our statefulset. You may chose to delete one or both stateful sets, depending on which database volumes you wish to expand. The `--cascade=false` flag is most important here.
```

Comment:
```suggestion
In this step we will delete the stateful set, which controls our database pod. The command below will delete the referenced database's stateful set without deleting the pod. We do not want to delete the pod itself as this would cause downtime. In the following steps we will be redeploying our stateful set. You might chose to delete one or both stateful sets, depending on which database volumes you wish to expand. The `--cascade=false` flag is most important here.
```